src/Microsoft.VisualStudio.Services.Agent/SecretMasker.cs

```diff
@@ -99,31 +99,108 @@ public string MaskSecrets(string input)
             return stringBuilder.ToString();
         }
 
-        public void Add(ISecret mask)
+        public void AddRegex(string expression)
         {
-            _secretMasks.Add(mask);
+            _secrets.Add(new RegexSecret(expression));
         }
 
-        public void AddRegEx(string expression)
+        public void AddValue(string value)
         {
-            Add(new RegexSecret(expression));
+            if (!string.IsNullOrEmpty(value))
+            {
+                _secrets.Add(new ValueSecret(value));
+            }
         }
 
-        public void AddValue(string value)
+        // TODO: REVISIT WRT CONCURRENCY. NEED TO SWAP OUT UNDERLYING LIST TO AVOID INTERFERENCE WITH ITERATORS.
+        public void AddVariable(string name, string value)
         {
-            Add(new ValueSecret(value));
+            _variableNames.Add(name);
+            if (!string.IsNullOrEmpty(value))
+            {
+                _secrets.Add(new ValueSecret(value));
```

Comment:
I removed the `VariableSecret` class since the Name property was never used. I switched this method to simply add a `ValueSecret` to the list instead.